src/PhpSpreadsheet/Settings.php

```diff
@@ -166,23 +161,19 @@ public static function getCache()
         return self::$cache;
     }
 
-    /**
-     * Set the HTTP client implementation to be used for network request.
-     */
-    public static function setHttpClient(ClientInterface $httpClient): void
-    {
-        self::$client = $httpClient;
-    }
-
     /**
      * Get the HTTP client implementation to be used for network request.
      */
-    public static function getHttpClient(): ClientInterface
+    public static function getDefaultHttpClient(): ClientInterface
     {
-        if (!self::$client) {
-            self::$client = new Client();
+        if (class_exists(Client::class)) {
+            // Support for Guzzle
+            return new Client();
+        } elseif (class_exists(HttpClient::class)) {
+            // Support for Symfony HttpClient
+            return HttpClient::create();
```

Comment:
Let's provide support for at least 2 clients, just in case 😉

Reply:
This won't do. You removed dependencies on implementation in composer.json, but you re-add add them here in code. All references to specific implementations must disappear.

Reply:
Well, these are **optional** dependencies, since `class_exists()` is used, so if the class doesn't exist, it will fall back to an exception, telling the user to either inject an http client, or install one of the two supported implementations.

I forgot to do it in the PR, but I was about to add a `suggest` section in `composer.json` to invite users to install either `symfony/http-client` or `guzzlehttp/guzzle` to benefit from the `WEBSERVICE` feature.

This is kinda like automatic feature-flag depending on what composer packages are installed in the project